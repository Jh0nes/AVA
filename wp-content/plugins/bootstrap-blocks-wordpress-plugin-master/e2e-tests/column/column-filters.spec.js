/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	deactivatePlugin,
	createNewPost,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';
import {
	insertRowBlock,
} from '../row/row-helper';
import {
	selectColumnBlock,
} from './column-helper';
import {
	getCheckboxValueByLabel,
	getDataValuesOfElement,
	getInputValueByLabel,
	getSelectedValueBySelectLabel,
	openSidebarPanelWithTitle,
	selectOption,
	selectOptionIsAvailable,
} from '../helper';

describe( 'column block filters', () => {
	beforeAll( async () => {
		await activatePlugin( 'wp-bootstrap-blocks-test-column-filters' );
	} );

	afterAll( async () => {
		await deactivatePlugin( 'wp-bootstrap-blocks-test-column-filters' );
	} );

	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'wpBootstrapBlocks.column.bgColorOptions should add background color', async () => {
		await insertRowBlock();
		await selectColumnBlock();
		await openSidebarPanelWithTitle( 'Background color' );

		// Additional background color should be available
		expect( await page.$( '.components-color-palette__item[aria-label="Color: brand"]' ) ).not.toBeNull();

		// Background color should be applied
		await page.click( '.components-color-palette__item[aria-label="Color: brand"]' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'wpBootstrapBlocks.column.paddingOptions should add padding option', async () => {
		await insertRowBlock();
		await selectColumnBlock();
		await openSidebarPanelWithTitle( 'Padding (inside column)' );

		// Additional padding option should be available
		expect( await selectOptionIsAvailable( 'Size', 'p-8' ) ).toBe( true );

		// Padding option should be applied
		await selectOption( 'Size', 'p-8' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'wp_bootstrap_blocks_column_default_attributes should override default attributes', async () => {
		await insertRowBlock();
		await selectColumnBlock();

		// Columm size values should be set
		await openSidebarPanelWithTitle( 'Column size' );
		expect( await getInputValueByLabel( 'Xs Column count' ) ).toMatch( '4' );
		expect( await getInputValueByLabel( 'Sm Column count' ) ).toMatch( '6' );
		// For the md column count we would expect a value of 8 but it gets overwritten by the default layout which has a value of 6.
		expect( await getInputValueByLabel( 'Md Column count' ) ).toMatch( '6' );
		expect( await getInputValueByLabel( 'Lg Column count' ) ).toMatch( '10' );
		expect( await getInputValueByLabel( 'Xl Column count' ) ).toMatch( '0' );

		// Columm equal-width checkboxes should be checked
		expect( await getCheckboxValueByLabel( 'Xl equal-width' ) ).toBe( true );

		// Background color should be selected
		await openSidebarPanelWithTitle( 'Background color' );
		// There is no way to see which color of a color palette is selected. That's why we check the data attribute value.
		const columnData = await getDataValuesOfElement( '[data-type="wp-bootstrap-blocks/column"]' );
		expect( columnData.bgColor ).toMatch( 'primary' );
		expect( await getCheckboxValueByLabel( 'Center content vertically in row' ) ).toBe( true );

		// Padding should be selected
		await openSidebarPanelWithTitle( 'Padding (inside column)' );
		expect( await getSelectedValueBySelectLabel( 'Size' ) ).toMatch( 'p-3' );

		// Check if attributes are set correctly
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
