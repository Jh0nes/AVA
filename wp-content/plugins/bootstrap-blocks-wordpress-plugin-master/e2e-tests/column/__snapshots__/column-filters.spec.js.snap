// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`column block filters wp_bootstrap_blocks_column_default_attributes should override default attributes 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`column block filters wpBootstrapBlocks.column.bgColorOptions should add background color 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6,\\"bgColor\\":\\"brand\\"} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`column block filters wpBootstrapBlocks.column.paddingOptions should add padding option 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6,\\"padding\\":\\"p-8\\"} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;
